app/src/main/java/org/mozilla/fenix/ext/String.kt

```diff
@@ -113,17 +116,23 @@ fun String.simplifiedUrl(): String {
 }
 
 /**
- * Gets a rounded drawable from a URL if possible, else null. Must be called off main thread.
+ * Gets a rounded drawable from a URL if possible, else null.
  */
-fun String.decodeUrlToRoundedDrawable(context: Context): RoundedBitmapDrawable? {
-    val avatarUrl = try {
-        URL(this)
-    } catch (e: MalformedURLException) {
-        return null
+suspend fun String.toRoundedDrawable(context: Context, client: Client) = bitmapForUrl(this, client)?.let { bitmap ->
+    RoundedBitmapDrawableFactory.create(context.resources, bitmap).also {
+        it.isCircular = true
+        it.setAntiAlias(true)
     }
-    val bitmap = BitmapFactory.decodeStream(avatarUrl.openConnection().getInputStream())
-    val roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(context.resources, bitmap)
-    roundedBitmapDrawable.isCircular = true
-    roundedBitmapDrawable.setAntiAlias(true)
-    return roundedBitmapDrawable
+}
+
+suspend fun bitmapForUrl(url: String, client: Client): Bitmap? = withContext(Dispatchers.IO) {
+    // TODO cache this image, see https://github.com/mozilla-mobile/fenix/issues/9531
+    // Code below will cache it in Gecko's cache, which ensures that as long as we've fetched it once,
```

Comment:
When a user changes their account photo desktop, will we get a new URL? aka will this update or just keep using the cache?

Reply:
Not sure, I'll test this.

Reply:
Works correctly after app restart, when we refetch the profile.

After a profile change we have a push message coming in, telling the app that the profile changed; once we support it, we can refetch the profile without having to wait for a restart.